### PR TITLE
Extend wallet service

### PR DESCRIPTION
**Pull request by xitronix**

### CODE REVIEW

File: universal-login-wallet/src/services/WalletService.ts [hunk 1]
```diff
@@ -1,16 +1,47 @@
...
+  }
+
+  async createFutureWallet(): Promise<FutureWallet> {
+    const futureWallet = await this.sdk.createFutureWallet();
+    this.setFutureWallet(futureWallet);
+    return futureWallet;
+  }
+
+  setFutureWallet(userWallet: FutureWallet) {
+    this.state = 'Future';
```
Comment (marekkirejczyk) [inline universal-login-wallet/src/services/WalletService.ts]: `ensure(this.state == null)`

File: universal-login-wallet/src/services/WalletService.ts [hunk 2]
```diff
@@ -1,16 +1,47 @@
...
-  public userWallet?: UserWallet | null;
-  walletExists = () => !!this.userWallet;
+  public userWallet?: FutureWallet | UserWallet;
+  public state: WalletState = 'None';
+
+  constructor(private sdk: UniversalLoginSDK) {
+  }
+
+  walletExists(): boolean {
+    return !!this.userWallet;
```
Comment (marekkirejczyk) [inline universal-login-wallet/src/services/WalletService.ts]: `return this.state != null`

File: universal-login-wallet/src/services/WalletService.ts [hunk 3]
```diff
@@ -1,16 +1,47 @@
...
+    this.setFutureWallet(futureWallet);
+    return futureWallet;
+  }
+
+  setFutureWallet(userWallet: FutureWallet) {
+    this.state = 'Future';
+    this.userWallet = userWallet;
+  }
+
+  setUserWallet(userWallet: UserWallet) {
```
Comment (marekkirejczyk) [inline universal-login-wallet/src/services/WalletService.ts]: `ensure(this.state == 'Future')`

File: universal-login-wallet/src/services/WalletService.ts [hunk 4]
```diff
@@ -1,16 +1,47 @@
...
+    this.state = 'Deployed';
+    this.userWallet = userWallet;
+  }
 
-  disconnect(): any {
-    this.userWallet = null;
+  disconnect(): void {
```
Comment (marekkirejczyk) [inline universal-login-wallet/src/services/WalletService.ts]: `ensure(this.state != 'None');`

File: universal-login-wallet/src/services/WalletService.ts (hunk 3 above)
Comment (xitronix) [reply universal-login-wallet/src/services/WalletService.ts]: ✔️

File: universal-login-wallet/src/services/WalletService.ts (hunk 1 above)
Comment (xitronix) [reply universal-login-wallet/src/services/WalletService.ts]: ✔️

File: universal-login-wallet/src/services/WalletService.ts (hunk 2 above)
Comment (xitronix) [reply universal-login-wallet/src/services/WalletService.ts]: Changed to `this.state === 'Deployed'`

**Pull request merged**